matter_server/server/storage.py

```diff
@@ -0,0 +1,142 @@
+"""Logic to handle storage of persistent data."""
+
+import asyncio
+import logging
+import os
+import pprint
+from types import NoneType
+from typing import TYPE_CHECKING, Any, Dict, Union
+
+from genericpath import isfile
+
+from ..common.helpers.json import JSON_DECODE_EXCEPTIONS, json_dumps, json_loads
+
+DEFAULT_SAVE_DELAY = 30
+
+if TYPE_CHECKING:
+    from .server import MatterServer
+
+
+class StorageController:
+    """Controller that handles storage of persistent data."""
+
+    def __init__(self, server: "MatterServer") -> None:
+        """Initialize storage controller."""
+        self.server = server
+        self.logger = logging.getLogger(__name__)
```

Comment:
We could use a constant at the module level instead as we're just using the module logger here. We don't seem to need an instance attribute.